Odd test failure in otel-collector

```
$ make test
skipping go generate ./...
cd ./cmd/opentelemetry-collector && bash -c "set -e; set -o pipefail; go test -v -race ./... | sed ''/PASS/s//PASS/'' | sed ''/FAIL/s//FAIL/''"

...

ok  	github.com/jaegertracing/jaeger/cmd/opentelemetry-collector/app/exporter/cassandra	2.989s
=== RUN   TestDefaultConfig
--- PASS: TestDefaultConfig (0.02s)
=== RUN   TestLoadConfigAndFlags
--- PASS: TestLoadConfigAndFlags (0.03s)
=== RUN   TestCreateTraceExporter
    TestCreateTraceExporter: factory_test.go:41:
        	Error Trace:	factory_test.go:41
        	Error:      	Error message not equal:
        	            	expected: "failed to create primary Elasticsearch client: health check timeout: Head http://127.0.0.1:9200: dial tcp 127.0.0.1:9200: connect: connection refused: no Elasticsearch node available"
        	            	actual  : "failed to create primary Elasticsearch client: health check timeout: Head \"http://127.0.0.1:9200\": dial tcp 127.0.0.1:9200: connect: connection refused: no Elasticsearch node available"
        	Test:       	TestCreateTraceExporter
--- FAIL: TestCreateTraceExporter (5.03s)
=== RUN   TestCreateTraceExporter_nilConfig
--- PASS: TestCreateTraceExporter_nilConfig (0.00s)
=== RUN   TestCreateMetricsExporter
--- PASS: TestCreateMetricsExporter (0.00s)
=== RUN   TestCreateDefaultConfig
--- PASS: TestCreateDefaultConfig (0.00s)
=== RUN   TestType
--- PASS: TestType (0.00s)
FAIL
FAIL	github.com/jaegertracing/jaeger/cmd/opentelemetry-collector/app/exporter/elasticsearch	8.397s
```

cc @pavolloffay

--- a/cmd/opentelemetry-collector/app/defaults/default_config_test.go
+++ b/cmd/opentelemetry-collector/app/defaults/default_config_test.go
@@ -126,7 +126,7 @@ func TestDefaultCollectorConfig(t *testing.T) {
 			cfg, err := CollectorConfig(test.storageType, test.zipkinHostPort, factories)
 			if test.err != "" {
 				require.Nil(t, cfg)
-				assert.EqualError(t, err, test.err)
+				assert.Contains(t, err.Error(), test.err)
 				return
 			}
 			require.NoError(t, err)

--- a/cmd/opentelemetry-collector/app/exporter/cassandra/factory_test.go
+++ b/cmd/opentelemetry-collector/app/exporter/cassandra/factory_test.go
@@ -38,14 +38,14 @@ func TestCreateTraceExporter(t *testing.T) {
 	}}
 	exporter, err := factory.CreateTraceExporter(context.Background(), component.ExporterCreateParams{}, factory.CreateDefaultConfig())
 	require.Nil(t, exporter)
-	assert.EqualError(t, err, "gocql: unable to create session: control: unable to connect to initial hosts: dial tcp 127.0.0.1:9042: connect: connection refused")
+	assert.Contains(t, err.Error(), "gocql: unable to create session")
 }
 
 func TestCreateTraceExporter_NilConfig(t *testing.T) {
 	factory := Factory{}
 	exporter, err := factory.CreateTraceExporter(context.Background(), component.ExporterCreateParams{}, nil)
 	require.Nil(t, exporter)
-	assert.EqualError(t, err, "could not cast configuration to jaeger_cassandra")
+	assert.Contains(t, err.Error(), "could not cast configuration to jaeger_cassandra")
 }
 
 func TestCreateDefaultConfig(t *testing.T) {

--- a/cmd/opentelemetry-collector/app/exporter/elasticsearch/factory_test.go
+++ b/cmd/opentelemetry-collector/app/exporter/elasticsearch/factory_test.go
@@ -38,14 +38,14 @@ func TestCreateTraceExporter(t *testing.T) {
 	}}
 	exporter, err := factory.CreateTraceExporter(context.Background(), component.ExporterCreateParams{}, factory.CreateDefaultConfig())
 	require.Nil(t, exporter)
-	assert.EqualError(t, err, "failed to create primary Elasticsearch client: health check timeout: Head http://127.0.0.1:9200: dial tcp 127.0.0.1:9200: connect: connection refused: no Elasticsearch node available")
+	assert.Contains(t, err.Error(), "failed to create primary Elasticsearch client")
 }
 
 func TestCreateTraceExporter_nilConfig(t *testing.T) {
 	factory := &Factory{}
 	exporter, err := factory.CreateTraceExporter(context.Background(), component.ExporterCreateParams{}, nil)
 	require.Nil(t, exporter)
-	assert.EqualError(t, err, "could not cast configuration to jaeger_elasticsearch")
+	assert.Contains(t, err.Error(), "could not cast configuration to jaeger_elasticsearch")
 }
 
 func TestCreateMetricsExporter(t *testing.T) {

--- a/cmd/opentelemetry-collector/app/exporter/grpcplugin/factory_test.go
+++ b/cmd/opentelemetry-collector/app/exporter/grpcplugin/factory_test.go
@@ -38,14 +38,14 @@ func TestCreateTraceExporter(t *testing.T) {
 	}}
 	exporter, err := factory.CreateTraceExporter(context.Background(), component.ExporterCreateParams{}, factory.CreateDefaultConfig())
 	require.Nil(t, exporter)
-	assert.EqualError(t, err, "error attempting to connect to plugin rpc client: fork/exec : no such file or directory")
+	assert.Contains(t, err.Error(), "error attempting to connect to plugin rpc client: fork/exec : no such file or directory")
 }
 
 func TestCreateTraceExporter_nilConfig(t *testing.T) {
 	factory := &Factory{}
 	exporter, err := factory.CreateTraceExporter(context.Background(), component.ExporterCreateParams{}, nil)
 	require.Nil(t, exporter)
-	assert.EqualError(t, err, "could not cast configuration to jaeger_grpc_plugin")
+	assert.Contains(t, err.Error(), "could not cast configuration to jaeger_grpc_plugin")
 }
 
 func TestCreateMetricsExporter(t *testing.T) {

--- a/cmd/opentelemetry-collector/app/exporter/kafka/factory_test.go
+++ b/cmd/opentelemetry-collector/app/exporter/kafka/factory_test.go
@@ -39,14 +39,14 @@ func TestCreateTraceExporter(t *testing.T) {
 	}}
 	exporter, err := factory.CreateTraceExporter(context.Background(), component.ExporterCreateParams{Logger: zap.NewNop()}, factory.CreateDefaultConfig())
 	require.Nil(t, exporter)
-	assert.EqualError(t, err, "kafka: client has run out of available brokers to talk to (Is your cluster reachable?)")
+	assert.Contains(t, err.Error(), "kafka: client has run out of available brokers to talk to (Is your cluster reachable?)")
 }
 
 func TestCreateTraceExporter_nilConfig(t *testing.T) {
 	factory := &Factory{}
 	exporter, err := factory.CreateTraceExporter(context.Background(), component.ExporterCreateParams{}, nil)
 	require.Nil(t, exporter)
-	assert.EqualError(t, err, "could not cast configuration to jaeger_kafka")
+	assert.Contains(t, err.Error(), "could not cast configuration to jaeger_kafka")
 }
 
 func TestCreateMetricsExporter(t *testing.T) {

--- a/cmd/opentelemetry-collector/app/exporter/span_writer_exporter_test.go
+++ b/cmd/opentelemetry-collector/app/exporter/span_writer_exporter_test.go
@@ -109,7 +109,7 @@ func TestStore(t *testing.T) {
 			dropped, err := test.storage.traceDataPusher(context.Background(), test.data)
 			assert.Equal(t, test.dropped, dropped)
 			if test.err != "" {
-				assert.EqualError(t, err, test.err)
+				assert.Contains(t, err.Error(), test.err)
 			} else {
 				require.NoError(t, err)
 			}

--- a/cmd/opentelemetry-collector/app/receiver/kafka/factory_test.go
+++ b/cmd/opentelemetry-collector/app/receiver/kafka/factory_test.go
@@ -30,7 +30,6 @@ import (
 	jConfig "github.com/jaegertracing/jaeger/pkg/config"
 )
 
-// TODO failing
 func TestCreateTraceReceiver(t *testing.T) {
 	v, _ := jConfig.Viperize(ingesterApp.AddFlags)
 	opts := DefaultOptions()
@@ -40,14 +39,14 @@ func TestCreateTraceReceiver(t *testing.T) {
 	}}
 	exporter, err := factory.CreateTraceReceiver(context.Background(), component.ReceiverCreateParams{Logger: zap.NewNop()}, factory.CreateDefaultConfig(), nil)
 	require.Nil(t, exporter)
-	assert.EqualError(t, err, "kafka: client has run out of available brokers to talk to (Is your cluster reachable?)")
+	assert.Contains(t, err.Error(), "kafka: client has run out of available brokers to talk to (Is your cluster reachable?)")
 }
 
 func TestCreateTraceExporter_nilConfig(t *testing.T) {
 	factory := &Factory{}
 	exporter, err := factory.CreateTraceReceiver(context.Background(), component.ReceiverCreateParams{}, nil, nil)
 	require.Nil(t, exporter)
-	assert.EqualError(t, err, "could not cast configuration to jaeger_kafka")
+	assert.Contains(t, err.Error(), "could not cast configuration to jaeger_kafka")
 }
 
 func TestCreateMetricsExporter(t *testing.T) {

